src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,22 @@
 package core.basesyntax;
 
-public class Car {
+public class Car implements Cloneable {
+    private String model;
+    private int numOfSits;
+    private Engine engine;
+    private String type;
+    private int numOfWheels;
 
+    public Car(String model, int numOfSits, Engine engine, String type, int numOfWheels) {
+        this.model = model;
+        this.numOfSits = numOfSits;
+        this.engine = engine;
+        this.type = type;
+        this.numOfWheels = numOfWheels;
+    }
+
+    @Override
+    protected Car clone() throws CloneNotSupportedException {
+        return new Car(model, numOfSits, engine, type, numOfWheels);
```

Comment:
you are adding the same engine object to Car constructor. What for you implemented clone() for Engine?